sevenseconds/config/vpc.py

```diff
@@ -622,6 +623,100 @@ def create_vpc_endpoints(account: AccountData, vpc: object, region: str):
+                    sg.authorize_ingress(
+                        IpProtocol='tcp',
+                        FromPort=443,
+                        ToPort=443,
+                        CidrIp='0.0.0.0/0',
+                    )
+                endpoints = ec2c.describe_vpc_endpoints(
+                    Filters=[
+                        {'Name': 'service-name', 'Values': [service_name]},
+                        {'Name': 'vpc-id', 'Values': [vpc.id]},
+                        {'Name': 'vpc-endpoint-state', 'Values': ['pending', 'available']},
+                    ]
+                )['VpcEndpoints']
+                if endpoints:
+                    for endpoint in endpoints:
+                        sgs_in_endpoint = [group['GroupId'] for group in endpoint['Groups']]
+                        if sg.id not in sgs_in_endpoint:
+                            options = {
+                                'VpcEndpointId': endpoint['VpcEndpointId'],
+                                'AddSecurityGroupIds': [sg.id],
+                            }
+                            act.warning(
+                                'mismatch ({} not in {}), make update: {}'.format(
+                                    sg.id,
+                                    sgs_in_endpoint,
+                                    ec2c.modify_vpc_endpoint(**options),
+                                )
+                            )
+                        if not endpoint['PrivateDnsEnabled']:
+                            options = {
+                                'VpcEndpointId': endpoint['VpcEndpointId'],
+                                'PrivateDnsEnabled': True,
+                            }
+                            act.warning(
+                                'mismatch (PrivateDns not enabled), make update: {}'.format(
+                                    ec2c.modify_vpc_endpoint(**options),
+                                )
+                            )
+                        subnet_in_endpoint = set(endpoint['SubnetIds'])
+                        if subnet_in_endpoint != subnets:
+                            options = {'VpcEndpointId': endpoint['VpcEndpointId']}
+                            if subnet_in_endpoint.difference(subnets):
+                                options['RemoveSubnetIds'] = list(
+                                    subnet_in_endpoint.difference(subnets)
+                                )
+                            if subnets.difference(subnet_in_endpoint):
+                                options['AddSubnetIds'] = list(
+                                    subnets.difference(subnet_in_endpoint)
+                                )
+                            act.warning(
+                                'mismatch ({} vs. {}), make update: {}'.format(
+                                    subnet_in_endpoint,
+                                    subnets,
+                                    ec2c.modify_vpc_endpoint(**options),
```

Comment:
I would move the modifications out of the log statement, otherwise it's very hard to notice the actual changes done.